File: src/main/java/net/sf/rails/game/model/PlayerNameModel.java
package net.sf.rails.game.model;

import net.sf.rails.game.Player;
import net.sf.rails.game.state.Model;

/**
 * PlayerNameModel contains both the name of the player
 * and information if the player has priority, thus it has to be observable
 *
 */
public class PlayerNameModel extends Model {
    
    public static final String ID = "PresidentModel";  
    
    private PlayerNameModel(Player player, String id) {
        super(player, id);
        // add dependency on the playerOrder
        player.getParent().getPlayerOrderModel().addModel(this);
    }

    public static PlayerNameModel create(Player player){
        return new PlayerNameModel(player, ID);
    }
    
    /**
     * @return restricted to Player
     */
    @Override
    public Player getParent() {
        return (Player)super.getParent();
    }
    
    @Override
    public String toText() {
        return getParent().getNameAndPriority();
    }

}

File: dotnet/RailsLib.Net/Net/Game/Model/PlayerNameModel.cs
using System;
using System.Collections.Generic;
using System.Text;

/**
 * PlayerNameModel contains both the name of the player
 * and information if the player has priority, thus it has to be observable
 *

 */
namespace GameLib.Net.Game.Model
{
    public class PlayerNameModel : State.Model
    {
        public const string ID = "PresidentModel";

        private PlayerNameModel(Player player, string id) : base(player, id)
        {
            // add dependency on the priority
            player.Parent.PriorityPlayerState.AddModel(this);
        }

        public static PlayerNameModel Create(Player player)
        {
            return new PlayerNameModel(player, ID);
        }

        /**
         * @return restricted to Player
         */
        new public Player Parent
        {
            get
            {
                return (Player)base.Parent;
            }
        }

        override public string ToText()
        {
            return Parent.GetNameAndPriority();
        }
    }
}
